src/mol-io/common/string-like.ts

```diff
@@ -0,0 +1,572 @@
+/**
+ * Copyright (c) 2025 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Adam Midlik <midlik@gmail.com>
+ */
+
+
+interface CustomString {
+    // TODO think about this extending full `String` interface, and impl classes just throwing `NotImplementedError` on more advanced methods -> less likely to break typing of dependant code (will only break runtime when running on big strings)
+    __string_like__: true,
+
+    readonly [index: number]: string;
+
+    // [Symbol.iterator]: () => StringIterator<string>;
+
+    // from @types/node/compatibility/indexable.d.ts
+
+    at(index: number): string | undefined;
+
+    // // from typescript/lib/lib.es5.d.ts
+
+    /** Returns a string representation of a string. */
+    toString(): string; // TODO make supersure this is implemented properly
+
+    // /**
+    //  * Returns the character at the specified index.
+    //  * @param pos The zero-based index of the desired character.
+    //  */
+    // charAt(pos: number): string;
+
+    /**
+     * Returns the Unicode value of the character at the specified location.
+     * @param index The zero-based index of the desired character. If there is no character at the specified index, NaN is returned.
+     */
+    charCodeAt(index: number): number;
+
+    // /**
+    //  * Returns a string that contains the concatenation of two or more strings.
+    //  * @param strings The strings to append to the end of the string.
+    //  */
+    // concat(...strings: string[]): string;
+
+    /**
+     * Returns the position of the first occurrence of a substring.
+     * @param searchString The substring to search for in the string
+     * @param position The index at which to begin searching the String object. If omitted, search starts at the beginning of the string.
+     */
+    indexOf(searchString: string, position?: number): number;
+
+    // /**
+    //  * Returns the last occurrence of a substring in the string.
+    //  * @param searchString The substring to search for.
+    //  * @param position The index at which to begin searching. If omitted, the search begins at the end of the string.
+    //  */
+    // lastIndexOf(searchString: string, position?: number): number;
+
+    // /**
+    //  * Determines whether two strings are equivalent in the current locale.
+    //  * @param that String to compare to target string
+    //  */
+    // localeCompare(that: string): number;
```

Comment:
For the final version I don't think we need to keep all these commented out functions.

Reply:
Yep. Removed.